### PR TITLE
Add proximity to grocery store feature

### PR DESCRIPTION
**Pull request by Damonamajor**
This adds three features categories of OSM data (greengrocer, supermarket, and wholesale) to the model pipeline. It selects non_NULL values of OSM points and polygons, and applies them to the dist_to_nearest_geography macro. It also queries OSM each year from 2014 to whatever the present day is, meaning that we should run this before modeling.

### CODE REVIEW

File: dbt/models/proximity/docs.md [hunk 1]
```diff
@@ -84,6 +84,14 @@ Distance from each PIN to the nearest golf course.
 **Primary Key**: `pin10`, `year`
 {% enddocs %}
 
+# dist_pin_to_grocery_store
+
+{% docs table_dist_pin_to_grocery_store %}
+Distance from each PIN to the nearest grocery_store.
```
Comment (dfsnow) [inline dbt/models/proximity/docs.md:90]:
```suggestion
Distance from each PIN to the nearest grocery store.
```

Comment (dfsnow) [reply dbt/models/proximity/docs.md:90]: I would also add the OSM tags here used to define grocery store.

File: dbt/models/proximity/proximity.dist_pin_to_grocery_store.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
+-- CTAS to create a table of distance to the nearest secondary road for each PIN
```
Comment (dfsnow) [inline dbt/models/proximity/proximity.dist_pin_to_grocery_store.sql:1]:
```suggestion
-- CTAS to create a table of distance to the nearest grocery store for each PIN
```

**Review by dfsnow — COMMENTED** [review]
Some small nitpicks and then this is set to go!

File: dbt/models/proximity/docs.md (hunk 1 above)
Comment (Damonamajor) [reply dbt/models/proximity/docs.md]: > I would also add the OSM tags here used to define grocery store.

I assume this references dbt/models/proximity/columns.md.

File: dbt/models/proximity/columns.md [hunk 3]
```diff
@@ -181,6 +181,31 @@ Nearest golf course ID, either Cook County ID or OSM ID.
 Golf course data sourced from Cook County GIS and OpenStreetMap
 {% enddocs %}
 
+## nearest_grocery_store_dist_ft
+
+{% docs column_nearest_grocery_store_dist_ft %}
+Nearest grocery store distance (feet).
+
+Grocery store data sourced from Cook County GIS and OpenStreetMap
```
Comment (dfsnow) [inline dbt/models/proximity/columns.md:189]:
```suggestion
Nearest grocery store distance (feet)
```

File: dbt/models/proximity/columns.md [hunk 4]
```diff
@@ -181,6 +181,31 @@ Nearest golf course ID, either Cook County ID or OSM ID.
...
+{% enddocs %}
+
+## nearest_grocery_store_name
+
+{% docs column_nearest_grocery_store_name %}
+Nearest grocery store name via OSM.
+
+Grocery store data sourced from Cook County GIS and OpenStreetMap
```
Comment (dfsnow) [inline dbt/models/proximity/columns.md:197]:
```suggestion
Nearest grocery store name via OSM
```

File: dbt/models/proximity/columns.md [hunk 5]
```diff
@@ -181,6 +181,31 @@ Nearest golf course ID, either Cook County ID or OSM ID.
...
+{% enddocs %}
+
+## nearest_grocery_osm_id
+
+{% docs column_nearest_grocery_store_osm_id %}
+Nearest grocery store osm_id number via OSM.
+
+Grocery store data sourced from Cook County GIS and OpenStreetMap
```
Comment (dfsnow) [inline dbt/models/proximity/columns.md:205]:
```suggestion
Nearest grocery store osm_id number via OSM
```

File: dbt/models/proximity/columns.md [hunk 6]
```diff
@@ -181,6 +181,31 @@ Nearest golf course ID, either Cook County ID or OSM ID.
...
+{% enddocs %}
+
+
```
Comment (dfsnow) [inline dbt/models/proximity/columns.md:208]:
```suggestion
```

File: dbt/models/proximity/docs.md [hunk 7]
```diff
@@ -84,6 +84,14 @@ Distance from each PIN to the nearest golf course.
 **Primary Key**: `pin10`, `year`
 {% enddocs %}
 
+# dist_pin_to_grocery_store
+
+{% docs table_dist_pin_to_grocery_store %}
+Distance from each PIN to the nearest grocery store.
+
+**Primary Key**: `pin10`, `year`
```
Comment (dfsnow) [inline dbt/models/proximity/docs.md:92]: **suggestion:** Let's add a quick sentence in here about which OSM tags were used to query the grocery stores.

File: dbt/models/spatial/docs.md [hunk 8]
```diff
@@ -162,6 +162,14 @@ data and tagged OpenStreetMap amenities.
 **Geometry:** `MULTIPOLYGON`
 {% enddocs %}
 
+# grocery_store
+
+{% docs table_grocery_store %}
+Grocery stores. Locations sourced OpenStreetMap amenities.
+
+**Geometry:** `POINT`
```
Comment (dfsnow) [inline dbt/models/spatial/docs.md:170]: **suggestion:** Likewise here, include the tags used to query the stores.

File: dbt/models/spatial/docs.md [hunk 9]
```diff
@@ -166,6 +166,8 @@ data and tagged OpenStreetMap amenities.
 
 {% docs table_grocery_store %}
 Grocery stores. Locations sourced OpenStreetMap amenities.
+OSM tags include supermarket, wholesale, and greengrocer.
+Only attributes with valid names are kept.
```
Comment (dfsnow) [inline dbt/models/spatial/docs.md:170]:
```suggestion
Grocery stores. Locations sourced from OpenStreetMap (OSM).

OSM tags include:

- `shop=supermarket`
- `shop=wholesale`
- `shop=greengrocer`
  
Only attributes with valid names are kept.
```

File: dbt/models/proximity/docs.md [hunk 10]
```diff
@@ -88,6 +88,8 @@ Distance from each PIN to the nearest golf course.
 
 {% docs table_dist_pin_to_grocery_store %}
 Distance from each PIN to the nearest grocery store.
+OSM tags include supermarket, wholesale, and greengrocer.
+Only attributes with valid names are kept.
```
Comment (dfsnow) [inline dbt/models/proximity/docs.md:92]:
```suggestion
Distance from each PIN to the nearest grocery store. Locations sourced from OpenStreetMap (OSM).

OSM tags include:

- `shop=supermarket`
- `shop=wholesale`
- `shop=greengrocer`
  
Only attributes with valid names are kept.
```

File: dbt/models/proximity/columns.md [hunk 11]
```diff
@@ -184,28 +184,21 @@ Golf course data sourced from Cook County GIS and OpenStreetMap
...
+Nearest grocery store name via OSM
 {% enddocs %}
 
 ## nearest_grocery_osm_id
 
 {% docs column_nearest_grocery_store_osm_id %}
-Nearest grocery store osm_id number via OSM.
-
-Grocery store data sourced from Cook County GIS and OpenStreetMap
+Nearest grocery store osm_id number via OSM
```
Comment (dfsnow) [inline dbt/models/proximity/columns.md:199]:
```suggestion
Nearest grocery store ID number via OSM
```

**Review by dfsnow — APPROVED** [review]
Minor changes then set to go!

**Pull request merged**